zarr/storage.py

```diff
@@ -446,7 +446,11 @@ def init_array(
 
     if store_version == 3 and "zarr.json" not in store:
         # initialize with default zarr.json entry level metadata
-        store["zarr.json"] = store._metadata_class.encode_hierarchy_metadata(None)  # type: ignore
+        # fmt: off
```

Comment:
That's the only way I have found to ensure `black` doesn't create lines > 100 characters, while not moving `# type: ignore[union-attr]` to an unrelated line (the last line with the parenthesis).

Suggestions welcome.

Reply:
the ongoing v3 effort will be stripping out a lot of the old v3 logic, so do whatever appeases the linter here, no matter how ugly it is